examples/viewer.py

```diff
@@ -263,15 +286,30 @@ def draw_event(
 
         keys = active_agent_id_and_sensor_name
 
-        self.sim._Simulator__sensors[keys[0]][keys[1]].draw_observation()
-        agent = self.sim.get_agent(keys[0])
-        self.render_camera = agent.scene_node.node_sensor_suite.get(keys[1])
-        self.debug_draw()
-        self.render_camera.render_target.blit_rgba_to_default()
-        mn.gl.default_framebuffer.bind()
-
-        # draw CPU/GPU usage data and other info to the app window
-        self.draw_text(self.render_camera.specification())
+        if self.enable_batch_renderer:
+            keyframes = []
+            for i in range(self.num_env):
+                # Generate keyframe
+                keyframes.append(self.sim[i].gfx_replay_manager.extract_keyframe())
+                # Copy sensor transforms
+                sensor_suite = self.sim[i]._sensors
+                for sensor_uuid, sensor in sensor_suite.items():
+                    transform = sensor._sensor_object.node.absolute_transformation()
+                    self.replay_renderer.set_sensor_transform(i, sensor_uuid, transform)
+                # Apply keyframe
+                self.replay_renderer.set_environment_keyframe(i, keyframes[i])
+                # Render
+                self.replay_renderer.render(mn.gl.default_framebuffer)
+        else:
+            self.sim[0]._Simulator__sensors[keys[0]][keys[1]].draw_observation()
+            agent = self.sim[0].get_agent(keys[0])
+            self.render_camera[0] = agent.scene_node.node_sensor_suite.get(keys[1])
+            self.debug_draw(0)
+            self.render_camera[0].render_target.blit_rgba_to_default()
+            # draw CPU/GPU usage data and other info to the app window
+            # TODO: Doesn't work with batch renderer.
```

Comment:
Should be fixed in https://github.com/facebookresearch/habitat-sim/pull/1947/commits/d025013ea6f00b9bb54e299de48d69e2a227bded.